google/gax/api_callable.py

```diff
@@ -382,6 +382,28 @@ def inner(*args, **kwargs):
     return inner
 
 
+def _merge_options_metadata(options, settings):
```

Comment:
Sorry, can you be more specific about "comming last for `x-goog-api-client`? Currently the _CallSettings.merge() logic does not specify anything about ordering of `x-goog-api-client`.

As for `construct_settings merges` it merges static "config" data (not specific for request). The new method merges kwargs portion of the options, which is request-specific.

About ordering. Sorry, in my sample I put headers in "worng" ordering. The code itself actually merges in a way that `x-goog-api-client` header goes last. But in any case, the order of headers should not matter, as long as x-goog-api-client is unique in request (should be the case for us):
[RFC2616 secion-4.2](https://tools.ietf.org/html/rfc2616#section-4.2):

> The order in which header fields with differing field names are received is not significant. However, it is "good practice" to send general-header fields first, followed by request-header or response-header fields, and ending with the entity-header fields.